WordPress/Classes/Services/PostService.m

```diff
@@ -161,16 +162,109 @@ - (void)loadMorePostsOfType:(NSString *)postType
                    forBlog:blog
                    options:options
                    success:^(NSArray *posts) {
-        [self.managedObjectContext performBlock:^{
-            [self mergePosts:posts ofType:postType forBlog:blog purgeExisting:NO completionHandler:success];
-        }];
-    } failure:^(NSError *error) {
-        if (failure) {
-            [self.managedObjectContext performBlock:^{
-                failure(error);
-            }];
+                       [self.managedObjectContext performBlock:^{
+                           [self mergePosts:posts ofType:postType withStatuses:nil forBlog:blog purgeExisting:NO completionHandler:success];
+                       }];
+                   } failure:^(NSError *error) {
+                       if (failure) {
+                           [self.managedObjectContext performBlock:^{
+                               failure(error);
+                           }];
+                       }
+                   }];
+}
+
+- (void)syncPostsOfType:(NSString *)postType
+           withStatuses:(NSArray *)postStatus
+                forBlog:(Blog *)blog
+                success:(void (^)(BOOL hasMore))success
+                failure:(void (^)(NSError *))failure
+{
+    NSString *status = [postStatus componentsJoinedByString:@","];
+    NSMutableDictionary *options = [NSMutableDictionary dictionary];
+    id<PostServiceRemote> remote = [self remoteForBlog:blog];
+    if ([remote isKindOfClass:[PostServiceRemoteREST class]]) {
+        options[@"status"] = status;
+    } else {
+        options[@"post_status"] = status;
+    }
+    options[@"number"] = @(PostServiceNumberToFetch);
+    [remote getPostsOfType:postType
+                   forBlog:blog
+                   options:options
+                   success:^(NSArray *posts) {
+                       BOOL hasMore = ([posts count] < PostServiceNumberToFetch) ? NO : YES;
+                       [self.managedObjectContext performBlock:^{
+                           [self mergePosts:posts ofType:postType withStatuses:postStatus forBlog:blog purgeExisting:YES completionHandler:^{
+                               if (success) {
```

Comment:
I've just checked the code, but just in case, double checking: this is *never* gonna be executed using a Background Context right?.

Because, if that's the case, the success callback would be running on a BG queue, and we just need to make sure we don't touch UIKit there.